src/mechanics/mesh/MeshGmsh.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#include "MeshFem.h"
+#include "base/Group.h"
+#include <string>
+#include <vector>
+#include <map>
+
+class GmshFileContent;
+
+namespace NuTo
+{
+
+//! @brief Reads a gmsh msh-file and converts it to a MeshFEM. Additionally physical groups are created and can be
+//! accessed via ID or name
+class MeshGmsh
```

Comment:
I am not sure how to handle the import from an API point of view. I would prefer a free function that somehow returns the functionality you provided. So 
~~~
auto gmshMesh = NuTo::ImportFromGmsh(filename);
MeshFem& mesh = gmshMesh.Mesh();
auto leftBoundary = gmshMesh.Get("LeftBoundary");
~~~
or something in that fassion.
The problem with that may be that the `MeshFem` is currently not copyable... :(

Reply:
I don't know. I don't see the benefit. Now you need a constructor to pass all the internal variables of MeshGmsh to the created mesh since the function can't access the internal variables of the mesh it returns until it is a friend function.

And... yeah, I just saw that we forgot to write a getter for the mesh...